pkg/tsdb/prometheus/resource/resource.go

```diff
@@ -81,45 +74,22 @@ func New(
 		return nil, fmt.Errorf("error reading settings: %w", err)
 	}
 
-	customHeaders := make(map[string]string)
 	var jsonDataMap map[string]interface{}
 
 	err := json.Unmarshal(settings.JSONData, &jsonDataMap)
 	if err != nil {
 		return nil, err
 	}
 
-	index := 1
-	for {
```

Comment:
This is just semi related and may be moved to separate PR. Passing the custom headers remove need for this, but the main reason we needed this is because we have a bug here https://github.com/grafana/grafana/blob/859cc92da70d5e37acc736031a325fc1479b772e/pkg/tsdb/prometheus/client/provider.go#L57 where we replace all the headers from the HTTP client setup code with just headers from the request. 

The question probably then is if adding custom headers to req make sense as we can then get it from 2 place (req or `HTTPClientOptions`) but I think it make sense to have it aligned with the query path.

Reply:
I'm planning to remove this kind of code as below everywhere as soon as Prometheus has been refactored :smile: 
https://github.com/grafana/grafana/blob/2705edf5f55fbfac4b9b6d588004bfa3dc923899/pkg/services/query/query.go#L187
https://github.com/grafana/grafana/blob/2705edf5f55fbfac4b9b6d588004bfa3dc923899/pkg/services/ngalert/eval/eval.go#L204

If this is required to make Prometheus in parity with query data/resource calls before Prometheus has been refactored I'm fine with it.

Reply:
Ah ok, I mean yeah removing the custom headers from query.go will also make this more aligned :D. So I think I will close this and fix this from the other end, fixing the HTTP client setup in Prometheus first.

Reply:
Yes, sounds good to me :+1: